integration/control.go

```diff
@@ -0,0 +1,13 @@
+package integration
+
+import v1 "github.com/juanfont/headscale/gen/go/headscale/v1"
+
+type ControlServer interface {
```

Comment:
Why are you not passing any Context in the interface?

Could help for timeouts...

Reply:
Currently it's all based in errors, so not getting a reply in time is built into the retry functionality and doesn't seem to care about context